### ISSUE TITLE
Animated gif not working

### ISSUE THREAD

**Issue by arhmart** [magento/magento2#38314]
### Preconditions and environment

Magento 2.4.7-beta2
Fresh install

### Steps to reproduce

1. Freshly install magento 2.4.7-beta2
2. Add new products
3. Upload animated GIF under the Images And Videos option
4. Save the product

### Expected result

Animated gif can play animation.

### Actual result

Animated gif cannot play animation.
I saw the introduction in the release notes of magento 2.4.7-beta2:

Merchants can now add a GIF image to a product’s image gallery from the Admin. Previously, Magento Open Source displayed this error: imagecolorsforindex(): Argument #2 ($color) is out of range.

Did I understand it wrong? Thanks for any help.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by arhmart** [comment]
@magento give me 2.4-develop instance

**Comment by arhmart** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @arhmart,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is reproducible.Kindly refer the attached video.


**Steps to reproduce**

- Freshly install magento 2.4.7-beta2
- Add new products
- Upload animated GIF under the Images And Videos option
- Save the product


[Screencast from 26-12-23 12:05:29 PM IST.webm](https://github.com/magento/magento2/assets/51680745/63a3ee5b-a822-4754-a98b-dda7aa667a48)


It is working fine in Admin.Animated gif cannot play animation in frontend.

Hence Confirming the issue.

Thanks.

**Comment by arhmart** [comment]
Can anyone tell me which file I need to modify to fix this? Thanks for any help!

**Comment by arhmart** [comment]
@engcom-Bravo Is this problem easy to solve? Is it possible to release a patch, my product contains a lot of animated gif. There are many Magento installations that desperately need this feature as well.

**Comment by arhmart** [comment]
Any help will be great.
@engcom-Bravo   @engcom-Dash   @engcom-November   @engcom-Hotel   @andrewbess   @manavluhar

**Comment by manavluhar** [comment]
@magento give me 2.4-develop instance

**Comment by manavluhar** [comment]
@arhmart , As I see, Images are getting resized to general magento image ratio and showing from cache directories. So for GIF. This may require some customisation in Magento/Catalog and in Magento/MediaStorage. 
I am curious and may look into it in more detail.
Thank you

**Comment by arhmart** [comment]
@manavluhar, Yes, as you can see, the gif doesn't work, but I don't know PHP. So I would like to ask if anyone who knows PHP can solve this issue.
Please let me know if there is any progress.
Thanks a lot.

**Comment by arhmart** [comment]
@engcom-Bravo, Thanks, I can't wait for the update.

**Comment by Jeffrey-H** [comment]
Is there any news on this issue?

**Comment by engcom-Hotel** [comment]
Hello @Jeffrey-H,

As of now, the JIRA indicates the status as `implementation`.

Following discussions on its corresponding JIRA, it's been determined that this should be categorized as a feature request rather than a bug. Therefore, we're relocating this issue to the feature request bucket.

Additionally, if the community develops this functionality, we'll gladly accept their contribution and incorporate this feature into our releases.

Thanks
